docs/tutorial/creating_cli.md

```diff
@@ -0,0 +1,622 @@
+
+The first function will add a contact to the contact list. But first, we need to define a couple of things: 
+
+The fist one will be the encoding of the `Contact` in the file. We’re going to go with a format like this: 
+
+```text
+<username>|<firstName>|<lastName>|<phoneNumber>|<email>
+```
+But of course, you can use any format you want, like JSON, CSV, etc.
+
+Once we defined the encoding, we need a function to parse the file representation to our type in Scala:
+
+```scala 3
+def impl(bookPath: Path): ContactManager = new ContactManager:
+  
+    private def parseContact(contact: String): Either[Exception, Contact] =
+      contact.split('|') match 
+        case Array(id, firstName, lastName, phoneNumber, email) =>
+          Contact(id, firstName, lastName, phoneNumber, email).asRight
+        case _ => new Exception(s"Invalid contact format: $contact").asLeft
+    end parseContact
+```
+
+Next, a function to encode the contact to a string:
+
+```scala 3
+    private def encodeContact(contact: Contact): String =
+      s"${contact.username}|${contact.firstName}|${contact.lastName}|${contact.phoneNumber}|${contact.email}"
+```
+
+Now, we can implement the `addContact` function:
+
+```scala 3
+override def addContact(contact: Contact): IO[Username] = 
+  bookPath.readLines.flatMap: lines =>               // (1)
+    lines.find(_.startsWith(contact.username)) match // (2)
+      case Some(_) =>
+        ContactFound(contact.username).raiseError[IO, Username]
+      
+      case None =>
+        lines.isEmpty
+          .pure[IO]
+          .ifM( // (3)
+            bookPath.write(encodeContact(contact)).as(contact.username),  // (4)
+            bookPath.appendLine(encodeContact(contact)).as(contact.username)
+          )
+          
+end addContact
+```
+
+First, we read all of our lines, line by line using the `FilesOs#readLines` method (1). Then, we check if the contact already exists in the file checking if the username is contained (2). If it does, we raise an error using the `raiseError` method. If it doesn't, we check if the file is empty (3) using the `ifM`; this method basically checks if the boolean inside the `IO` is true or false and it executes an effect depending on the condition. If it is, we write the contact to the file using the `write` method. If it's not, we append the contact to the file using the `appendLine` method (4).
+
+
+Now that we have the `addContact` function implemented, we can move to the `removeContact` function:
+
+```scala 3
+override def removeContact(username: Username): IO[Unit] =
+  for
+    contacts <- bookPath.readLines
+    _ <- bookPath.writeLines(
+      contacts.filterNot(_.startsWith(username))
+    )
+  yield ()
+```
+
+The explanation is straightforward. We load all the contacts from the file, and then we filter the ones that don't match the username we want to remove. Finally, we write the filtered contacts back to the file.
```

Comment:
> Either way, in this case it's redundant, because if the explanation is straightforward, might as well not mention it :P.

Lol, didn't see it like that, guess you are right :s